github/data_source_github_organization_test.go

```diff
@@ -54,4 +55,69 @@ func TestAccGithubOrganizationDataSource(t *testing.T) {
 		})
 
 	})
+
+	t.Run("queries for an organization with archived repos", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+
+		config := fmt.Sprintf(`
+			resource "github_repository" "archived" {
+				name         = "tf-acc-archived-%s"
+				archived     = true
+		  	}
+
+			data "github_organization" "skip_archived" {
+				name = "%s"
+				ignore_archived_repos = true
+				depends_on = [
+					github_repository.archived,
+				]
+			}
+			data "github_organization" "all_repos" {
+				name = "%s"
+				ignore_archived_repos = false
+				depends_on = [
+					github_repository.archived,
+				]
+			}
+			
+			output "should_be_false" {
+				value = contains(data.github_organization.skip_archived.repositories, github_repository.archived.full_name)
+			}
+			output "should_be_true" {
+				value = contains(data.github_organization.all_repos.repositories, github_repository.archived.full_name)
+			}
+		`, randomID, testOrganization, testOrganization)
+
+		check := resource.ComposeTestCheckFunc(
+			resource.TestCheckOutput("should_be_false", "false"),
```

Comment:
@kfcampbell this test is based on the ones I found [here](https://github.com/ovh/terraform-provider-ovh/blob/95e83bb4df06b18b8cc5b92dcb42981e153535be/ovh/data_cloud_project_users_test.go#L44), where it appears that ``TestCheckOutput()`` handles the conversions of bools and strings, i.e., `false` --> `"false"`. That code uses `terraform-plugin-sdk/v2`, while the GitHub provider uses `v1`. Is there a reason for not using v2 here?

I think I read somewhere in the docs that you guys don't want the developer community to not mess with bumping dependencies ourselves, which is why I'm asking 😄

Reply:
The reason is due to lack of prioritization of #1780, not anything super intentional. I'm fine adding v2 as a dependency since we'll need to cut over to it anyways.

Reply:
So how do we move forward here? Should I wait until #1780 is merged, or import and use `v2` in `data_source_github_organization_test.go` (if that is even possible, haven't dealt much with golang dependencies and vendoring before)? I don't care about getting this in fast, so if it takes another couple of weeks for #1780, I'm fine to wait

Reply:
Let's move forward with your second option. The approach you describe is indeed possible and I think it's unlikely #1780 is going to be merged in the near future.

Reply:
I played around a bit with option 2, but I think using both v1 and v2 of the sdk is not compatible, at least not without doing some real weird stuff. 

First of you need to use a version of the sdk before `v2.10.1` (which released in dec 2021). Using a later version requires a bump of `terraform-exec`, to a version above `v0.16.0`, [which deprecated the `tfinstall` package](https://github.com/hashicorp/terraform-exec/blob/f8daf3dc8e047bd082f80a0918488f06e3ce233b/CHANGELOG.md?plain=1#L79C1-L99) that v1 of the sdk depends on

I just go stuck with the tests using the [`testAccProviders` (defined in `github/provider_test.go`)](https://github.com/integrations/terraform-provider-github/blob/52489e144754805407365053e0c15f620ce8f39a/github/provider_test.go#L12-L14), which uses v1. They cannot be used for the v2 tests in `github/data_source_github_organization_test.go`.

Reply:
Oh gross...thanks for doing the legwork on that! I'm :+1: on waiting for now.

Reply:
If you're interested in picking up #1780 and fixing the build errors, please feel free! You should be able to create a new PR off of my branch if you desire.

Reply:
@kfcampbell that thought did cross my mind, especially since the time I spent working around it could have been enough to just make it work properly. I learnt a lot about go dependencies at least 😅 

I make take a look later

Reply:
Haha, I'm glad you were able to take something positive out of it! Should you end up deciding to do so, I'd be happy to look at it.

Reply:
With the new release this actually works, I've tested it locally.